packages/meteor-gazelle-ipmanager/lib/methods.js

```diff
@@ -0,0 +1,138 @@
+IpManager = {
+  validateLogin: function (allowed, ipAddr) {
+    return Meteor.call('ipmanager/validateLogin', allowed, ipAddr);
+  },
+  ip2long: function (ipAddr) {
+    var i = 0;
+    ipAddr = ipAddr.match(
+      /^([1-9]\d*|0[0-7]*|0x[\da-f]+)(?:\.([1-9]\d*|0[0-7]*|0x[\da-f]+))?(?:\.([1-9]\d*|0[0-7]*|0x[\da-f]+))?(?:\.([1-9]\d*|0[0-7]*|0x[\da-f]+))?$/i
+    );
+    if (!ipAddr) {
+      return false;
+    }
+
+    ipAddr[0] = 0;
+    for (i = 1; i < 5; i += 1) {
+      ipAddr[0] += !! ((ipAddr[i] || '')
+        .length);
+      ipAddr[i] = parseInt(ipAddr[i]) || 0;
+    }
+
+    ipAddr.push(256, 256, 256, 256);
+
+    ipAddr[4 + ipAddr[0]] *= Math.pow(256, 4 - ipAddr[0]);
+    if (ipAddr[1] >= ipAddr[5]
+        || ipAddr[2] >= ipAddr[6]
+        || ipAddr[3] >= ipAddr[7]
+        || ipAddr[4] >= ipAddr[8]) {
+      return false;
+    }
+    return ipAddr[1] * (ipAddr[0] === 1 || 16777216) + ipAddr[2] * (ipAddr[0] <= 2 || 65536) + ipAddr[3] * (ipAddr[0] <= 3 || 256) + ipAddr[4] * 1;
+  }
+};
+
+Meteor.methods({
+  'ipmanager/validateLogin': function (allowed, ipAddr) {
+    if (Meteor.call('ipmanager/ipIsBanned', ipAddr)) {
+      throw new Meteor.Error(403, Meteor.settings.USER_BANNED_ERRORMSG);
```

Comment:
It might be pertinent to move to a different object or file. But we should have a singular place to put error messaging wouldnt you say? Especially for ease of localization